web/hooks/use-group-supabase.ts

```diff
@@ -0,0 +1,196 @@
+import { JSONContent } from '@tiptap/core'
+import { User } from 'common/user'
+import { useEffect, useState } from 'react'
+import { groupRoleType } from 'web/components/groups/group-member-modal'
+import { db } from 'web/lib/supabase/db'
+import {
+  getGroupMembers,
+  getGroupOfRole,
+  getMemberRole,
+  getNumGroupMembers,
+  MEMBER_LOAD_NUM,
+} from 'web/lib/supabase/group'
+import { useAdmin } from './use-admin'
+import { useUser } from './use-user'
+
+export function useRealtimeRole(groupId: string | undefined) {
+  const [userRole, setUserRole] = useState<groupRoleType | null>(null)
+  const user = useUser()
+  const isManifoldAdmin = useAdmin()
+  useEffect(() => {
+    setTranslatedMemberRole(groupId, isManifoldAdmin, setUserRole, user)
+  }, [user])
+  useEffect(() => {
+    const channel = db.channel('user-group-role-realtime')
+    channel.on(
+      'postgres_changes',
+      {
+        event: '*',
+        schema: 'public',
+        table: 'group_members',
+        filter: `group_id=eq.${groupId}`,
+      },
+      (payload) => {
+        if (payload.new.member_id === user?.id) {
+          setTranslatedMemberRole(groupId, isManifoldAdmin, setUserRole, user)
+        }
+      }
+    )
+    channel.subscribe(async (status) => {})
+  }, [db, user])
```

Comment:
Shouldn't the hook return a function to unsubscribe from the channel?

`return () => db.removeChannel(channel)`

Reply:
u rite